aikau/src/main/resources/alfresco/layout/HorizontalWidgets.js

```diff
@@ -128,6 +138,70 @@ define(["alfresco/core/ProcessWidgets",
+       * that are returned by calling [getVisibilityRuleTopics]{@link module:alfresco/layout/HorizontalWidgets#getVisibilityRuleTopics}
+       * on the first pass through the [doWidthProcessing]{@link module:alfresco/layout/HorizontalWidgets#doWidthProcessing}
+       * function. The subscriptions need to be created after the widgets have been created in order that their visibility 
+       * is adjusted before the [onResize]{@link module:alfresco/layout/HorizontalWidgets#onResize} function that is bound 
+       * to is called.
+       *
+       * @instance
+       * @param {object[]} widgets The widgets that have been created
+       * @since 1.0.33
+       */
+      allWidgetsProcessed: function alfresco_layout_HorizontalWidgets__allWidgetsProcessed(/*jshint unused:false*/ widgets) {
+         this.inherited(arguments);
+         if (this.visibilityRuleTopics)
+         {
+            array.forEach(this.visibilityRuleTopics, function(topic) {
+               this.alfSubscribe(topic, lang.hitch(this, this.onResize));
+            }, this);
+         }
+      },
+
+      /**
+       * This function is called on the first pass through the 
+       * [doWidthProcessing]{@link module:alfresco/layout/HorizontalWidgets#doWidthProcessing} function and checks all
+       * the supplied widgets for dynamic visibility configuration so that subscriptions can be created
+       * on the same rules to trigger resizing as widgets are displayed or hidden.
+       * 
+       * @instance
+       * @param {object[]} widgets The widgets to check for visibility/invisibility configuration
+       * @return {string[]} An array of the topics that are using in dynamic visibility/invisibility configuration
+       * @since 1.0.33
+       */
+      getVisibilityRuleTopics: function alfresco_layout_HorizontalWidgets__getVisibilityRuleTopics(widgets) {
+         var topics = [];
+         var ruleTopicsObject = {};
+         array.forEach(widgets, function(widget) {
+            var visibilityRules = lang.getObject("config.visibilityConfig.rules", false, widget);
+            array.forEach(visibilityRules, function(rule) {
+               if (rule.topic && !ruleTopicsObject[rule.topic])
+               {
+                  ruleTopicsObject[rule.topic] = true;
+               }
+            });
+            var invisibilityRules = lang.getObject("config.invisibilityConfig.rules", false, widget);
+            array.forEach(invisibilityRules, function(rule) {
+               if (rule.topic && !ruleTopicsObject[rule.topic])
+               {
+                  ruleTopicsObject[rule.topic] = true;
+               }
+            });
+         });
+
+         for (var key in ruleTopicsObject) 
+         {
+            if (ruleTopicsObject.hasOwnProperty(key))
+            {
+               topics.push(key);
+            }
+         }
+         return topics;
```

Comment:
This is a very verbose function. It could be written more concisely ...

```javascript
var topicNames = {};
array.forEach(widgets, function(widget) {
  var visibilityRules = lang.getObject("config.visibilityConfig.rules", false, widget) || [],
    invisibilityRules = lang.getObject("config.invisibilityConfig.rules", false, widget) || [];
  array.forEach(visibilityRules.concat(invisibilityRules), function(rule) {
    if (rule.topic) {
      topicNames[rule.topic] = true;
    }
  });
});
return Object.keys(topicNames);
```

... if one wanted to, of course!